tools/crm_resource_ban.c

```diff
@@ -446,16 +440,20 @@ cli_resource_clear_all_expired(xmlNode *root, cib_t *cib_conn, int cib_options,
         xmlNode *constraint_node = getXpathResult(xpathObj, i);
         xmlNode *date_expr_node = NULL;
         crm_time_t *end = NULL;
-        char *xpath_string = NULL;
 
-        xpath_string = build_clear_xpath_string(constraint_node, rsc, node, promoted_role_only);
-        if (xpath_string == NULL) {
+        if (buf == NULL) {
+            buf = g_string_sized_new(1024);
+        }
```

Comment:
Alternatively could do the allocation inside `build_clear_xpath_string()` via a `GString **` so that it's only allocated if needed. It ought to be a tad more efficient but is a tad uglier. With this being just a CLI tool, I don't see it making much difference.